modules/ca/src/tools/caget.c

```diff
@@ -434,11 +436,15 @@ int main (int argc, char *argv[])
             enumAsNr = 1;
             break;
         case 'w':               /* Set CA timeout value */
-            if(epicsScanDouble(optarg, &caTimeout) != 1)
             {
-                fprintf(stderr, "'%s' is not a valid timeout value "
-                        "- ignored. ('caget -h' for help.)\n", optarg);
-                caTimeout = DEFAULT_TIMEOUT;
+                double prevTimeout = caTimeout;
+                if(epicsScanDouble(optarg, &caTimeout) != 1)
+                {
+                    caTimeout = prevTimeout;
```

Comment:
`epicsScanDouble()` (aka. `!epicsParseDouble()`) will only write to the output pointer on success.  So I don't think there it is necessary to explicitly save/restore the default timeout.